packages/blade/src/components/TopNav/TabNav/TabNavItem.web.tsx

```diff
@@ -0,0 +1,152 @@
+    }),
+    flex: 1,
+    display: 'flex',
+    gap: makeSpace(theme.spacing[2]),
+    alignItems: 'center',
+    justifyContent: 'center',
+    height: '100%',
+    textDecoration: 'none',
+    whiteSpace: 'nowrap',
+    paddingTop: makeSpace(theme.spacing[3]),
+    paddingBottom: makeSpace(theme.spacing[3]),
+    paddingLeft: makeSpace(theme.spacing[4]),
+    paddingRight: makeSpace(theme.spacing[4]),
+    borderRadius: makeBorderSize(theme.border.radius.medium),
+    '&:hover': $isActive
+      ? {}
+      : {
+          backgroundColor: theme.colors.interactive.background.gray.default,
+        },
+  };
+});
+
+const StyledTabNavItemWrapper = styled(BaseBox)<{ isActive?: boolean }>(({ theme, isActive }) => {
+  const dividerHiderStyle = {
+    content: '""',
+    position: 'absolute',
+    top: '50%',
+    transform: 'translateY(-50%)',
+    width: 2,
+    height: '80%',
+    backgroundColor: theme.colors.surface.background.gray.subtle,
+  } as const;
+
+  return {
+    position: 'relative',
+    flexShrink: 0,
+    padding: `${makeSpace(theme.spacing[2])} ${makeSpace(theme.spacing[1])}`,
+    backgroundColor: isActive ? theme.colors.surface.background.gray.intense : 'transparent',
+    borderColor: isActive ? theme.colors.surface.border.gray.muted : 'transparent',
+    borderStyle: 'solid',
+    borderBottomWidth: 0,
+    borderWidth: makeBorderSize(theme.border.width.thin),
+    borderTopLeftRadius: makeBorderSize(theme.border.radius.medium),
+    borderTopRightRadius: makeBorderSize(theme.border.radius.medium),
+    // Animation
+    transform: isActive ? 'translateY(2px)' : 'none',
+    transition: `${makeMotionTime(theme.motion.duration.moderate)} ${
+      theme.motion.easing.standard.effective
+    }`,
+    transitionProperty: 'background, transform',
+
+    // Hide the left and right divider by overlaying them with a pseudo element as same color as the background
+    ...(isActive
+      ? {
+          ':before, :after': dividerHiderStyle,
+          ':before': {
+            left: -3,
+          },
+          ':after': {
+            right: -3,
+          },
+        }
+      : {}),
+  };
+});
+
+const SelectedBar = styled(BaseBox)<{ isActive?: boolean }>(({ theme, isActive }) => {
+  return {
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    right: 0,
+    height: makeSpace(theme.spacing[1]),
+    borderTopLeftRadius: makeBorderSize(theme.border.radius.medium),
+    borderTopRightRadius: makeBorderSize(theme.border.radius.medium),
+    backgroundColor: theme.colors.interactive.icon.gray.normal,
+    pointerEvents: 'none',
+    // Animation
+    opacity: isActive ? 1 : 0,
+    transition: `${makeMotionTime(theme.motion.duration.moderate)} ${
+      theme.motion.easing.standard.effective
+    }`,
+    transitionProperty: 'opacity',
+  };
+});
+
+const _TabNavItem: React.ForwardRefRenderFunction<HTMLAnchorElement, TabNavItemProps> = (
+  {
+    as,
+    children,
+    isActive,
+    leading: Leading,
+    trailing: Trailing,
```

Comment:
Missed it in API decision. Shouldn't leading and trailing be JSX instead of Component? that is what we have in other items 🤔

Reply:
Fixed, changed to icon/trailing